sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/EmptyNullableValueAccess.CSharp9.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+
+int? topLevel = 42;
+var v = topLevel.Value;
+
+topLevel = null;
+v = topLevel.Value;   // FN
```

Comment:
For this rule we should add a test with target typed conditional. Something similar with:
```
int? a = null;
int b = 2;

var x = true ? a : b;
var y = x.Value;
```

Reply:
That's not a target-typed conditional, but a normal conditional. I couldn't find a scenario with target typed conditional for this rule :(

Reply:
We could add though the following:
```
ICollection<string> collection = condition
                                ? new List<string>()
                                : new LinkedList<string>();
collection.Clear();

var list = new List<string>() { "a"};
var linkedList = new LinkedList<string>();
linkedList.AddLast("b");

ICollection<string> c = condition ? list : linkedList; 
c.Clear();
```
in `EmptyCollectionsShouldNotBeEnumerated.CSharp9.cs`. This is for sure target-type conditional 😄